core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -478,11 +493,20 @@ object AdminClient {
       retryBackoffMs,
       requestTimeoutMs.toLong)
 
-    new AdminClient(
+    val adminClient = new AdminClient(
       time,
       requestTimeoutMs,
       retryBackoffMs,
       highLevelClient,
       bootstrapCluster.nodes.asScala.toList)
+
+    // wait until an necessary authentication is performed
+    while (adminClient.getNonRetriableException == null)
+      Thread.sleep(retryBackoffMs)
+
+    if (adminClient.getNonRetriableException.isDefined)
+      throw adminClient.getNonRetriableException.get
+
```

Comment:
This is the reason for the transient timeout test failure. This waits for a successful connection, which causes the first request to succeed in the test. This block of code is not required. We should simply propagate authentication exceptions from request failures.